store/tikv/client.go

```diff
@@ -172,6 +306,158 @@ func (a *connArray) Close() {
 	close(a.streamTimeout)
 }
 
+type batchCommandsEntry struct {
+	req *tikvpb.BatchCommandsRequest_Request
+	res chan *tikvpb.BatchCommandsResponse_Response
+
+	// Indicated the request is canceled or not.
+	canceled int32
+	err      error
+}
+
+// fetchAllPendingRequests fetches all pending requests from the channel.
+func fetchAllPendingRequests(
+	ch chan *batchCommandsEntry,
+	maxBatchSize int,
+	entries *[]*batchCommandsEntry,
+	requests *[]*tikvpb.BatchCommandsRequest_Request,
+) {
+	// Block on the first element.
+	headEntry := <-ch
```

Comment:
do we need to consider timeout here?

Reply:
Don't need. If there are no any requests come, the function should block here.